tests/js/client/shell/shell-debug-dump.js

```diff
@@ -0,0 +1,345 @@
+      for (let i = 0; i < 10; ++i) {
+        db[vn1].insert({_key: "value" + i + ":abc" + i, value: "value" + i});
+        db[vn2].insert({_key: "value" + i + ":def" + i, value: "value" + i});
+      }
+      for (let i = 0; i < 10; ++i) {
+        db[relationName].insert({
+          _from: vn1 + "/value" + i + ":abc" + i,
+          _to: vn2 + "/value" + i + ":def" + i,
+          _key: "value" + i + ":" + i + "123:" + "value" + i,
+          value: "value" + i
+        });
+      }
+      smartGraph._graph(graphName);
+      namedGraph._addVertexCollection(vn3);
+      db._createEdgeCollection(cn1);
+      db[cn1].save(`${vn1}/pet-dog`, `${vn2}/customer-abc`, {type: "belongs-to"});
+    },
+
+    tearDown: function () {
+      db._useDatabase("_system");
+      db._dropDatabase(dbName);
+      try {
+        fs.unlink(fileName);
+      } catch (err) {
+      }
+      try {
+        fs.unlink(outFileName);
+      } catch (err) {
+      }
+    },
+
+    testDebugDumpWithSmartGraphNamed: function () {
+      const graphBefore = db._graphs.document(graphName);
+      delete graphBefore._rev;
+      const query = "FOR v,e, p IN 0..10 ANY 'customer/customer-abc' GRAPH 'myGraph'  RETURN {key: e._key}";
+      explainer.debugDump(fileName, query, {}, {});
+      explainer.inspectDump(fileName, outFileName);
+      recreateEmptyDatabase();
+      internal.load(outFileName);
+      assertNull(db[vn1].any());
+      assertNull(db[vn2].any());
+      assertNull(db[relationName].any());
+      const graphAfter = db._graphs.document(graphName);
+      delete graphAfter._rev;
+      assertEqual(graphBefore, graphAfter, "graph recreated after debugDump is not the same as the original");
+    },
+
+    testDebugDumpWithSmartGraphNamedWithExamples: function () {
+      let vn1Before = [];
+      db[vn1].all().toArray().forEach(vertex => {
+        vn1Before.push(vertex._key);
+      });
+      vn1Before.sort();
+      let vn2Before = [];
+      db[vn2].all().toArray().forEach(vertex => {
+        vn2Before.push(vertex._key);
+      });
+      vn2Before.sort();
+      let relationBefore = [];
+      db[relationName].all().toArray().forEach(edge => {
+        relationBefore.push(edge._key);
+      });
+      relationBefore.sort();
+      const graphBefore = db._graphs.document(graphName);
+      delete graphBefore._rev;
+      const query = "FOR v,e, p IN 0..10 ANY 'customer/customer-abc' GRAPH 'myGraph'  RETURN {key: e._key}";
+      explainer.debugDump(fileName, query, {}, {examples: 10});
+      explainer.inspectDump(fileName, outFileName);
+      recreateEmptyDatabase();
+      internal.load(outFileName);
+      const graphAfter = db._graphs.document(graphName);
+      delete graphAfter._rev;
+      assertEqual(graphBefore, graphAfter, "graph recreated after debugDump is not the same as the original");
+      let vn1After = [];
+      db[vn1].all().toArray().forEach(vertex => {
+        vn1After.push(vertex._key);
+      });
+      vn1After.sort();
```

Comment:
This extraction code is duplicated several times.
Would make sense to put this into a reusable function.